ENH: Add run_stale_examples config var (#759)
sphinx-gallery/sphinx-gallery: doc/configuration.rst

        $ sphinx-build -D sphinx_gallery_conf.filename_pattern=plot_specific_example\.py ...

.. _run_stale_examples:

Rerunning stale examples
========================
By default, sphinx-gallery only rebuilds examples that have changed.

> For example, starting from a clean ``doc/`` directory, running your HTML build

Reviewer: ```suggestion
For example, when starting from a clean ``doc/`` directory, running your HTML build
```
?

> once will result in Sphinx-gallery executing all examples that match your given

Author: There appear to be 15 of each kind in `configuration.rst`, plus 30 for Sphinx-Gallery. Not sure if we have a preference. Let's fix all of these in another PR

> filename / ignore patterns. Then, running

Reviewer: :ref:`<filename / ignore patterns> build_pattern`

the exact same command a second time *should not run any examples*, because the
MD5 hash of each example will be checked against the MD5 hash (saved to disk
as ``<filename>.md5`` in the generated directory) that the example file had

> during the first build, they will match and thus the example will be determined

Reviewer: Sentence is a little long. Maybe:
```suggestion
during the first build. These will match and thus the example will be determined
```

to be "stale", and it will not be rebuilt by Sphinx-gallery.
This design feature allows for more rapid documentation iteration by only
rebuilding examples when they change.

However, this presents a problem during some modes of debugging and
iteration. Let's say that you have one particular
example that you want to rebuild repeatedly while modifying some function in

> your underlying library, while leaving the example file contents themselves
> fixed. To do this, you'd either need to make some change (e.g., add/delete a

Reviewer: ```suggestion
your underlying library but do not want to change the example file contents
themselves. To do this, you'd either need to make some change (e.g., add/delete a
```

newline) to your example or delete the ``.md5`` file to force Sphinx-gallery
to rebuild the example. Instead, you can use the configuration value::

    sphinx_gallery_conf = = {
        ...
        'run_stale_examples': True,
    }

> 

Reviewer: With this configuration all examples will be rebuilt, even if their MD5 hash show that the examples have not changed.

> This can be combined with other options to repeatedly rerun a single example

Reviewer: ```suggestion
You can combine this configuration with :ref:<filename / ignore patterns> build_pattern to repeatedly rerun a single example.
```

> from the command line for example by doing some variant of:

Reviewer: ```suggestion
This could be done from the command line, for example:
```

.. code-block:: console

    $ make html SPHINXOPTS='-D sphinx_gallery_conf.run_stale_examples=True' -Dsphinx_gallery_conf.filename_pattern='my_example_name'``

> This will cause any examples matching the filename/ignore patterns to be

Reviewer: ```suggestion
This command will cause any examples matching the filename pattern 'my_example_name' to be
```
